src/modules/utils/directory.rs

```diff
@@ -0,0 +1,23 @@
+/// Truncate a path to only have a set number of path components
```

Comment:
Good idea splitting this out into a utility 👍 
I think it would be best if we also moved the associated unit tests to this file:
https://github.com/Scotsguy/starship/blob/fix-conda-long-paths/src/modules/directory.rs#L238-L278